apps/dash-label-properties/app.py

```diff
@@ -0,0 +1,357 @@
+import dash_html_components as html
+import dash_core_components as dcc
+import dash_bootstrap_components as dbc
+from dash.exceptions import PreventUpdate
+import dash_table
+from dash_table.Format import Format
+import plotly.express as px
+import plotly.graph_objs as go
+
+import numpy as np
+from skimage import io, filters, measure
+import pandas as pd
+
+import PIL
+from skimage import color, img_as_ubyte
+from plotly import colors
+from textwrap import dedent
+
+
+external_stylesheets = [dbc.themes.BOOTSTRAP]
+
+app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
+server = app.server
+
+# Read the image that will be segmented
+filename = (
+    "https://upload.wikimedia.org/wikipedia/commons/a/ac/Monocyte_no_vacuoles.JPG"
+)
+img = io.imread(filename, as_gray=True)[:660:2, :800:2]
+labels = measure.label(img < filters.threshold_otsu(img))
+props = measure.regionprops(labels, img)
+
+# Define table columns
+list_columns = [
+    "label",
+    "area",
+    "perimeter",
+    "eccentricity",
+    "euler_number",
+    "mean_intensity",
+]
+columns = [
+    {"name": label_name, "id": label_name}
+    if precision is None
+    else {
+        "name": label_name,
+        "id": label_name,
+        "type": "numeric",
+        "format": Format(precision=precision),
+    }
+    for label_name, precision in zip(list_columns, (None, None, 4, 4, None, 3))
+]
+table = pd.DataFrame(
+    [[getattr(prop, col) for col in list_columns] for prop in props],
+    columns=list_columns,
+)
+
+
+def image_with_contour(img, labels, mode="lines", shape=None):
+    """
+    Figure with contour plot of labels superimposed on background image.
+
+    Parameters
+    ----------
+
+    img : URL, dataURI or ndarray
+        Background image. If a numpy array, it is transformed into a PIL
+        Image object.
+    labels : 2D ndarray
+        Contours are the isolines of labels.
+    shape: tuple, optional
+        Shape of the arrays, to be provided if ``img`` is not a numpy array.
+    """
+    try:
+        sh_y, sh_x = shape if shape is not None else img.shape
+    except AttributeError:
+        print(
+            """the shape of the image must be provided with the
+                 ``shape`` parameter if ``img`` is not a numpy array"""
+        )
+    if type(img) == np.ndarray:
+        img = img_as_ubyte(color.gray2rgb(img))
+        img = PIL.Image.fromarray(img)
+    labels = labels.astype(np.float)
+    custom_viridis = colors.PLOTLY_SCALES["Viridis"]
+    custom_viridis.insert(0, [0, "#FFFFFF"])
+    custom_viridis[1][0] = 1.0e-4
+    # Contour plot of segmentation
+    print("mode is", mode)
+    opacity = 0.4 if mode is None else 1
+
+    fig = px.imshow(img, binary_backend="jpg")
+    fig.add_contour(
+        z=labels,
+        contours=dict(start=0, end=labels.max() + 1, size=1, coloring=mode),
+        line=dict(width=1),
+        showscale=False,
+        colorscale=custom_viridis,
+        opacity=opacity,
```

Comment:
a hovertemplate could be added here to display only the `z` value (not `x` and `y`) and to add for example the mean intensity (passed as `customdata`, see for example https://plotly.com/python/hover-text-and-formatting/#adding-other-data-to-the-hover-with-customdata-and-a-hovertemplate)